ui/apps/platform/src/Containers/ReduxUserPermissionProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import type { Access } from 'types/role.proto';
+import type { ResourceName } from 'types/roleResources';
+import { selectors } from 'reducers';
+import { createStructuredSelector } from 'reselect';
+import { UserPermissionContext } from 'hooks/usePermissions';
+
+const stateSelector = createStructuredSelector<{
+    userRolePermissions: { resourceToAccess: Partial<Record<ResourceName, Access>> };
```

Comment:
nit: I found the following in the redux selector (`lines 58-63` in `roles.js`):

```
const userRolePermissions = (state = null, action) => {
    if (action.type === types.FETCH_USER_ROLE_PERMISSIONS.SUCCESS) {
        return isEqual(action.response, state) ? state : action.response;
    }
    return state;
};
```

Since the state is initially `null` should we change this to `Partial<Record<ResourceName, Access>> | null` to be more correct? Seems like we do handle the `null` case during runtime. I thought we could make the type more correct as well.

Reply:
Great catch - will update.